### PR TITLE
Implemented send, receiveTab, copyToMy, changePassword, withdraw. Changed search little bit

### PR DESCRIPTION
**Pull request by Tommybro**
Now user can send time table to another user (can't send to himself)
Implemented all about receiveTab
=> show myTimeTable
=> show receivedTimeTables
=> show unreadReceivedTimeTables
=> show sender of receivedTimeTables
Can copy timeTable to myTimeTable
Can change user's password (when change password, automatically sign out)
Can withdraw (user can delete himself)

Changed course.name.contains to course.name.abbrev when search lecture

### CODE REVIEW

File: src/components/molecules/TimeTable/index.js [hunk 1]
```diff
@@ -3,9 +3,10 @@ import Button from '../../atoms/Button'
 import LecturePopup from '../LecturePopup'
 
 
-const TimeTable = ({ id, memo, title, lectures, canDeleteLecture, onModifyContent, onDeleteLecture, onBookmark }) => {
+const TimeTable = ({ username, id, memo, title, lectures, canModify, canCopyToMy, onModifyContent, onDeleteLecture, onBookmark, onSendTimeTable, onCopyToMy }) => {
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: `onBookmark`, `onSendTimeTable`, and `onCopyToMy`.
I think they should share same naming policy.
`onBookmarkTimeTable`, `onSendTimeTable`, `onCopyToMyTimeTable`
Or,
`onBookmark`, `onSend`, `onCopyToMy`.

**Comment by holenet** [comment]
Can I see the screenshots of (maybe)four tabs?

**Comment by Tommybro** [comment]
recommend
bookmark
receivedtt
settings

File: src/store/ttrs/reducer.js [hunk 2]
```diff
@@ -153,6 +154,33 @@ const timeTable = (state = [], action) => {
...
+        ...state,
+        receivedTimeTables: action.receivedTimeTables.map((timeTable) => ({
+          ...timeTable,
+        })),
+        receivedTimeTable: action.receivedTimeTables.length === 0 ? initialTimeTable.receivedTimeTable : action.receivedTimeTables[0],
+      }
+    case actions.SELECT_RECEIVED_TIME_TABLE_RESPONSE:
+      receivedTimeTables = state.receivedTimeTables.map((timeTable) => ({
+        ...timeTable,
+      }))
```
Comment (holenet) [inline src/store/ttrs/reducer.js]: These three lines above can be replaced simply with this one line?
```js
receivedTimeTables = [...state.receivedTimeTables]
```

Comment (Tommybro) [reply src/store/ttrs/reducer.js]: OH it works

File: src/store/ttrs/sagas.js [hunk 3]
```diff
@@ -92,6 +110,7 @@ function* signIn(username, password) {
   try {
     const response = yield call(axios.get, 'ttrs/students/my/', config)
     console.log('signIn response', response)
+    response.data.password = password
```
Comment (holenet) [inline src/store/ttrs/sagas.js]: Holding plain raw password of the user seems not good as the frontend server.
I know this is for checking old password input in changing password.
But, there is one more way to check that.
Just re-sign in with that old password. If successful, the password that the user has typed turns out to be the right password.

Comment (Tommybro) [reply src/store/ttrs/sagas.js]: But.. why holding plain raw password of the user seems not good?

Comment (holenet) [reply src/store/ttrs/sagas.js]: Actually, I'm not sure about that point.
I just thought so.
I have searched several articles about that problem.
But, I could not get a good answer.
I was just concerned about whether the state of redux is vulnerable to attacks from the outside.
I'll devise a method to increase security level of both the backend and the frontend.

Comment (Tommybro) [reply src/store/ttrs/sagas.js]: OK~

File: src/store/ttrs/actions.js [hunk 4]
```diff
@@ -288,3 +289,10 @@ export const deleteTimeTable = (timeTableId, timeTableType) => {
     timeTableType,
   }
 }
+
+export const deleteMyTimeTable = (myTimeTable) => {
+  return {
+    type: DELETE_MY_TIME_TABLE,
+    myTimeTable,
```
Comment (holenet) [inline src/store/ttrs/actions.js]: Is it necessary?

Comment (Tommybro) [reply src/store/ttrs/actions.js]: No

Comment (holenet) [reply src/store/ttrs/actions.js]: Delete it.

Comment (Tommybro) [reply src/store/ttrs/actions.js]: Did it

**Pull request merged**